rllib/utils/test_utils.py

```diff
@@ -748,22 +753,24 @@ def should_check_eval(experiment):
             # Experiment finished: Check reward achieved and timesteps done
             # (throughput).
             else:
+                # Use best_result's reward to check min_reward.
                 if check_eval:
                     episode_reward_mean = np.mean(
                         [
-                            t.last_result["evaluation"]["episode_reward_mean"]
+                            t.metric_analysis["evaluation/episode_reward_mean"]["max"]
```

Comment:
the max of the reward_mean .... hmm what does this mean?

Reply:
it's actually the mean of the max :)
but we only run 1 trial for each test, so it's basically just the max reward for that run I think.

Reply:
wait doesn't this change the interpretation of this significantly before -- effectively lowering the threshold?

Reply:
oh I mis-understood your question. sorry.
so the idea is this, we have been checking whether a Trial is successful or not by using the last data point.
but the last data point may not be the best data point of a training run. you can have overfitting toward the end, and have significantly worse policy by then.
so this change is to basically say:

I am gonna take the best reward this run has achieved, and see if that passes my threshold.

do you think this is a better idea??

Reply:
Yeah, this makes perfect sense now. I like this change. Should give us more stability w/o(!) lowering our  reward threshold.